frigate/detectors/plugins/tensorrt.py

```diff
@@ -115,41 +134,78 @@ def _allocate_buffers(self):
-            # Allocate host and device buffers
-            err, host_mem = cuda.cuMemHostAlloc(
-                nbytes, Flags=cuda.CU_MEMHOSTALLOC_DEVICEMAP
-            )
-            assert err is cuda.CUresult.CUDA_SUCCESS, f"cuMemAllocHost returned {err}"
-            logger.debug(
-                f"Allocated Tensor Binding {binding} Memory {nbytes} Bytes ({size} * {self.engine.get_binding_dtype(binding)})"
-            )
-            err, device_mem = cuda.cuMemAlloc(nbytes)
-            assert err is cuda.CUresult.CUDA_SUCCESS, f"cuMemAlloc returned {err}"
-            # Append the device buffer to device bindings.
-            bindings.append(int(device_mem))
-            # Append to the appropriate list.
-            if self.engine.binding_is_input(binding):
-                logger.debug(f"Input has Shape {binding_dims}")
-                inputs.append(HostDeviceMem(host_mem, device_mem, nbytes, size))
+                assert err is cuda.CUresult.CUDA_SUCCESS, f"cuMemAllocHost returned {err}"
+                logger.debug(
+                    f"Allocated Tensor Binding {binding} Memory {nbytes} Bytes ({size} * {self.engine.get_binding_dtype(binding)})"
+                )
+                err, device_mem = cuda.cuMemAlloc(nbytes)
+                assert err is cuda.CUresult.CUDA_SUCCESS, f"cuMemAlloc returned {err}"
+                # Append the device buffer to device bindings.
+                bindings.append(int(device_mem))
+                # Append to the appropriate list.
+                if self.engine.binding_is_input(binding):
+                    logger.debug(f"Input has Shape {binding_dims}")
+                    inputs.append(HostDeviceMem(host_mem, device_mem, nbytes, size))
+                else:
+                    # each grid has 3 anchors, each anchor generates a detection
+                    # output of 7 float32 values
+                    assert size % 7 == 0, f"output size was {size}"
+                    logger.debug(f"Output has Shape {binding_dims}")
+                    outputs.append(HostDeviceMem(host_mem, device_mem, nbytes, size))
+                    output_idx += 1
             else:
-                # each grid has 3 anchors, each anchor generates a detection
-                # output of 7 float32 values
-                assert size % 7 == 0, f"output size was {size}"
-                logger.debug(f"Output has Shape {binding_dims}")
-                outputs.append(HostDeviceMem(host_mem, device_mem, nbytes, size))
-                output_idx += 1
+                binding_dims = self.engine.get_tensor_shape(binding)
+                if len(binding_dims) == 4:
+                    # explicit batch case (TensorRT 7+)
+                    size = trt.volume(binding_dims)
+                elif len(binding_dims) == 3:
+                    # implicit batch case (TensorRT 6 or older)
+                    size = trt.volume(binding_dims) * self.engine.max_batch_size
```

Comment:
Do can you remove this check if we already know TRT >= 10

Reply:
I did not realize that this would reduce supported cards, I am not sure we would want this

Reply:
This PR in itself doesn't, as the Frigate TensorRT image still uses v8 AND the all functions for tensorrt<=v8 have been kept. I simply added support for v10.

We could add a separate Frigate TensorRT image for v10, but it's not mandatory. And Nvidia hasn't yet released a docker image with TensorRT 10 (to replace `nvcr.io/nvidia/l4t-tensorrt:r8.5.2-runtime` in `workflows/ci.yml` and `nvcr.io/nvidia/tensorrt:23.03-py3` in `docker/tensorrt/Dockerfile.base`)

Reply:
@NateMeyer related PR https://github.com/NateMeyer/tensorrt_demos/pull/2